### ISSUE TITLE
优化器设置问题

### ISSUE THREAD

**Issue by Wangpeiyi9979** [circlePi/Bert_Chinese_Ner_pytorch#6]
您好，想请问一下，这几行关于调节学习率代码：
` 
if (step + 1) % args.gradient_accumulation_steps == 0:
                # modify learning rate with special warm up BERT uses
                lr_this_step = args.learning_rate * warmup_linear(global_step / t_total, args.warmup_proportion)
                for param_group in optimizer.param_groups:
                    param_group['lr'] = lr_this_step
                optimizer.step()
                optimizer.zero_grad()
                global_step += 1
`
用处大吗，有相关参考文献吗。。

**Comment by circlePi** [comment]
您好，这是梯度累积操作。一般情况下是一个batch更新一次梯度，这段代码可以gradient_accumulation_steps个batch更新一次梯度。用处是，比如我们的显存不够，batchsize只能设为2，但是这个batchsize太小了，这时候就可以通过gradient_accumulation_steps来变相增大batchsize了。

**Comment by circlePi** [comment]
参考文献的话，我记得有苏剑林的科学空间博客提过这个问题，可以去瞅瞅哈。
